test/e2e/sparkapplication_test.go

```diff
@@ -393,4 +395,86 @@ var _ = Describe("Example SparkApplication", func() {
 			Expect(strings.Contains(string(bytes), "Pi is roughly 3")).To(BeTrue())
 		})
 	})
+
+	Context("spark-pi: Suspend and Resume", func() {
+		ctx := context.Background()
+		path := filepath.Join("..", "..", "examples", "spark-pi.yaml")
+		app := &v1beta2.SparkApplication{}
+
+		BeforeEach(func() {
+			By("Parsing SparkApplication from file")
+			file, err := os.Open(path)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(file).NotTo(BeNil())
+
+			decoder := yaml.NewYAMLOrJSONDecoder(file, 100)
+			Expect(decoder).NotTo(BeNil())
+			Expect(decoder.Decode(app)).NotTo(HaveOccurred())
+
+			By("Creating SparkApplication")
+			Expect(k8sClient.Create(ctx, app)).To(Succeed())
+		})
+
+		AfterEach(func() {
+			key := types.NamespacedName{Namespace: app.Namespace, Name: app.Name}
+			Expect(k8sClient.Get(ctx, key, app)).To(Succeed())
+
+			By("Deleting SparkApplication")
+			Expect(k8sClient.Delete(ctx, app)).To(Succeed())
+		})
+
+		It("should complete successfully", func() {
+			key := types.NamespacedName{Namespace: app.Namespace, Name: app.Name}
+
+			By("Waiting for SparkApplication to Running")
+			Eventually(func(g Gomega) {
+				app := &v1beta2.SparkApplication{}
+				g.Expect(k8sClient.Get(ctx, key, app)).To(Succeed())
+				g.Expect(app.Status.AppState.State).To(Equal(v1beta2.ApplicationStateRunning))
+			}).WithTimeout(3 * time.Minute).Should(Succeed())
+			Consistently(func(g Gomega) {
+				app := &v1beta2.SparkApplication{}
+				g.Expect(k8sClient.Get(ctx, key, app)).To(Succeed())
+				g.Expect(app.Status.AppState.State).To(Equal(v1beta2.ApplicationStateRunning))
+			}).WithTimeout(5 * time.Second).Should(Succeed())
+
+			By("Suspending Spark Application")
+			Expect(k8sClient.Get(ctx, key, app)).To(Succeed())
+			app.Spec.Suspend = ptr.To(true)
+			Expect(k8sClient.Update(ctx, app)).To(Succeed())
+
+			By("Waiting for SparkApplication to Suspended")
+			Eventually(func(g Gomega) {
+				app := &v1beta2.SparkApplication{}
+				g.Expect(k8sClient.Get(ctx, key, app)).To(Succeed())
+				g.Expect(app.Status.AppState.State).To(Equal(v1beta2.ApplicationStateSuspended))
+			}).WithTimeout(3 * time.Minute).Should(Succeed())
+
+			By("Resuming for SparkApplication")
+			Expect(k8sClient.Get(ctx, key, app)).To(Succeed())
+			app.Spec.Suspend = ptr.To(false)
+			Expect(k8sClient.Update(ctx, app)).To(Succeed())
+
+			By("Waiting for SparkApplication to Running")
+			Eventually(func(g Gomega) {
+				app := &v1beta2.SparkApplication{}
+				g.Expect(k8sClient.Get(ctx, key, app)).To(Succeed())
+				g.Expect(app.Status.AppState.State).To(Equal(v1beta2.ApplicationStateRunning))
+			}).WithTimeout(3 * time.Minute).Should(Succeed())
```

Comment:
How long it is typically in the Running state? Asking because this check is intrinsically flaky - if the app computes very quickly you may not notice the "Running" state, just "Completed" immediately. Note that "Eventually" performs periodic checks. Still, if the "running" lasts for >3s then it is very unlikely to flaky.

Reply:
Sure, got it, then I'm fine keeping this check. Feel free to resolve the comment.